src/assets/sql/stored-procedure/getAppointmentWithUserIdAndServiceIdAndPatientId.sql

```diff
@@ -0,0 +1,94 @@
+-------------------------------------------------------------------------- STORED PROCEDURE SCRIPT----------------------------------------------------------------
+DELIMITER $$
+DROP PROCEDURE IF EXISTS `getAppointmentsByConditionAndPaging` $$
+CREATE PROCEDURE `getAppointmentsByConditionAndPaging`(IN `idUserIntegrated` VARCHAR(200), 
+                                                       IN `idServiceIntegrated` VARCHAR(200),
+                                                       IN `idPatientIntegrated` VARCHAR(200), 
+                                                       IN `startDate` VARCHAR(200), 
+                                                       IN `endDate` VARCHAR(200),
+                                                       IN `page` INT, 
+                                                       IN `size` INT, 
+                                                       IN `sort` VARCHAR(200))
+BEGIN
+	Declare startOffset Integer;
+    #-- Prepare pre-condition before running main query
+    IF idServiceIntegrated <> '' OR idServiceIntegrated IS NOT NULL THEN
+        SET idServiceIntegrated = (SELECT ea_services.id from `ea_services` WHERE ea_services.id_integrated = idServiceIntegrated);
+    END IF;
+
+    IF idUserIntegrated <> '' OR idUserIntegrated IS NOT NULL THEN
+        SET idUserIntegrated = (SELECT ea_users.id from ea_users WHERE ea_users.id_integrated = idUserIntegrated);
+    END IF;
+
+    IF idPatientIntegrated <> '' OR idPatientIntegrated IS NOT NULL THEN
+        SET idPatientIntegrated = (SELECT ea_users.id from ea_users WHERE ea_users.id_integrated = idPatientIntegrated);
+    END IF;
+
+    IF page <= 0 OR page IS NULL THEN
+          SET page = 1; #default page = 1
+    END IF;
+
+    IF size <= 0 OR size IS NULL THEN
+        SET size = 10; #default size = 10
+    END IF;
+
+    IF sort IS NULL OR sort <> 'ASC' OR sort <> 'DESC' THEN
+        SET sort = 'ASC'; #default size = 10
+    END IF;
+
+    SET startOffset = ((page - 1 ) * size);
+    
+
+    # -- Start build final Query with parameter
+    SET @finalQuery = 'SELECT *  FROM ea_appointments WHERE 1 = 1';
```

Comment:
remove dummy WHERE

Reply:
could not remove WHERE 1 =1 , this 1 =1 is a trick to concentrate multiple condition base on parameter. In case you don't all null parameter, you can be able to get all records with pagination.